Added Azure.ContainerApp.Name

BenjaminEngeset opened the pull request:
## PR Summary

Fixes #2094 

Added Azure.ContainerApp.Name rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

BernieWhite commented on src/PSRule.Rules.Azure/rules/Azure.ContainerApp.Rule.yaml:
```diff
@@ -27,6 +27,29 @@ spec:
+      - name: '.'
+        lessOrEquals: 32
+      - name: '.'
+        match: '^([a-z][0-9a-z-]{1,30}[0-9]|[a-z][0-9])$'
```
Maybe try this expression instead.

```suggestion
        match: '^([a-z][0-9a-z-]{0,30}[0-9a-z])$'
```

BernieWhite reviewed (changes requested):
Nice work @BenjaminEngeset. Just a few points to tweak this further. Let me know if you have any questions.

BenjaminEngeset commented:
@BernieWhite looks like it didn't work, but looks like you are aware of it.

BernieWhite commented on tests/PSRule.Rules.Azure.Tests/Azure.ContainerApp.Tests.ps1:
```diff
@@ -71,3 +71,53 @@ Describe 'Azure.ContainerApp' -Tag 'ContainerApp' {
+
+        $testObject = [PSCustomObject]@{
+            Name = ''
+            ResourceType = 'Microsoft.Resources/deployments'
```
Needs to be the same type as the rule.

```suggestion
            ResourceType = 'Microsoft.App/containerApps'
```

BernieWhite reviewed (changes requested):
I added an issue for future support for `caseSensitive` of `match`.

The rule test is failed for another reason, see below.

BenjaminEngeset commented:
> I added an issue for future support for `caseSensitive` of `match`.
> 
> The rule test is failed for another reason, see below.

Great! Fixed btw.

BernieWhite reviewed (approved):
Thanks @BenjaminEngeset looks good to me. Should be good to merge.

BernieWhite reviewed (changes requested):
See suggestions below.

BernieWhite commented:
Adding that in.

BernieWhite reviewed (approved):
Thanks @BenjaminEngeset. All good to merge.
